StashRepository: Use StringUtils.isNotEmpty(), it can handle null

proski opened the pull request:
StringUtils handles `null` safely, which makes it possible to remove comparison to `null` and avoid some repetition.

`isNotEmpty` is much easier to read than the expression it replaces.

jakub-bochenski commented:
Do we by chance have any test coverage of this code?

jakub-bochenski commented:
I don't think there is any.
Could you please add some unit tests for the refactored methods, just to be on the safe side?

proski commented:
Will do

jakub-bochenski reviewed (changes requested):
See above

proski commented:
Unit tests have detected a pretty bad bug in the code, so I've submitted the fix as #81. This PR can be closed or kept around as a reminder that we could be using `StringUtils.equals` once the bigger issue is resolved.

I really don't want to add unit tests for incorrect functionality, and it's hard to avoid adding them to check this PR. In particular, pull requests with `getState()` returning `null` are considered buildable unconditionally.

proski commented:
I've removed the chunk that conflicts with #81, so this PR can be merged. If having unit tests is a requirement, then #81 should be merged first, and then I can add unit tests covering the changes in this PR.

jakub-bochenski commented:
#81 was merged but I don't think it covers the functionality changed here? At least not the skip phrase

proski commented:
> #81 was merged but I don't think it covers the functionality changed here? At least not the skip phrase

Correct, I'll need to add unit tests here.

proski committed:
StashRepositoryTest: Add unit tests for isForTargetBranch() and isSkipBuild()

proski committed:
StashRepository: Use StringUtils.isNotEmpty(), it treats null as empty

proski committed:
StashRepositoryTest: Set Mockito strictness to STRICT_STUBS

This makes unit tests warn about unnecessary mocking so the code is kept
short and to the point.

proski commented:
I have added unit tests for the affected code. The code changes are not intended to make any difference in the behavior, so the unit tests pass with and without those changes as expected.

I don't understand why the checks are failing here. I will force push later to force a recheck.

proski commented:
`java.io.IOException: Failed to create a new directory /tmp/jenkins37458170079657819tmp`
No, that's not from the new unit tests.

jakub-bochenski commented:
> No, that's not from the new unit tests.

Looks like an infra failure. Anyhow I think the last commit has passed

jakub-bochenski commented on src/test/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepositoryTest.java:
```diff
@@ -84,4 +95,156 @@ public void getTargetPullRequestsSkipsNullStatePullRequests() {
+
+  @Test
+  public void getTargetPullRequestsBuildsIfSkipPhraseIsNull() {
+    when(stashApiClient.getPullRequests()).thenReturn(Collections.singletonList(pullRequest));
```
there is a lot of repetition here, it would be nice to make it more DRY later

proski replied:
I can do it, but it would affect the existing tests, so let do it separately.

I'm considering constructing simple data objects like I do with the branch. It's should be more readable and more amenable to refactoring that mocking every trivial dependency. I could create objects in `before()` and use them in tests, changing them if necessary.

proski replied:
Done in #94.
